src/client/lcd/api/TxAPI.ts

```diff
@@ -545,7 +548,58 @@ export class TxAPI extends BaseAPI {
             TxInfo.fromData(tx_response)
           ),
           pagination: d.pagination,
+          total: Number(d.total),
         }
       })
   }
+
+  public async searchEvents(
+    moduleAddr: string,
+    moduleName: string,
+    startHeight: number,
+    endHeight: number
+  ): Promise<Event[]> {
+    if (endHeight < startHeight) {
+      throw new Error(`Start height cannot be greater than end height`)
+    }
+
+    if (endHeight - startHeight > 100) {
+      throw new Error(`Query height range cannot be greater than 100`)
+    }
+
+    const targetEvents: Event[] = []
+    const targetTag = `${moduleAddr}::${moduleName}`
+    const query: TxSearchQuery[] = [
+      { key: 'tx.height', value: startHeight.toString(), op: '>=' },
+      { key: 'tx.height', value: endHeight.toString(), op: '<=' },
+      { key: 'move.type_tag', value: targetTag, op: 'CONTAINS' },
+    ]
+    const filterEvents = (tx: TxInfo) => {
+      return tx.events.filter((event) => {
+        if (event.type !== 'move') return false
+        const typeTag = event.attributes.find((attr) => attr.key === 'type_tag')
+        return typeTag && typeTag.value.startsWith(targetTag)
+      })
+    }
+
+    const { txs, total } = await this.search({ query })
```

Comment:
it seems good to put 'pagination.limit': '100' and 'pagination.count_total': false explicitly.